extensions/omni_sql/omni_sql.c

```diff
@@ -583,19 +583,129 @@ Datum is_returning_statement(PG_FUNCTION_ARGS) {
 
 PG_FUNCTION_INFO_V1(execute_parameterized);
 
+/** Holds information between multiple calls to the SRF execute_parameterized function */
+typedef struct ExecCtx {
+  char *stmt;
+  // returning is important to see if a given statement returns tuples
+  // to the caller. This controls the flow of the main function
+  bool returning;
+  bool done;
+  Portal portal;
+  int nargs;
+  Oid *types;
+  char *nulls;
+  Datum *values;
+} ExecCtx;
+
+/*
+ * Extract types, values and nullability from the parameters array
+ * containing the values that should be used for a parameterized
+ * statement.
+ */
+void extract_information_json_parameters(ExecCtx *call_ctx, FuncCallContext *funcctx, Jsonb *parameters) {
```

Comment:
It would probably make sense it `static inline` so it 1) doesn't get exported 2) can get inlined